internal/services/network/web_application_firewall_policy_resource.go

```diff
@@ -680,8 +684,8 @@ func expandWebApplicationFirewallPolicyExclusions(input []interface{}) *[]networ
 	return &results
 }
 
-func expandWebApplicationFirewallPolicyManagedRuleSet(input []interface{}, d *pluginsdk.ResourceData) (*[]network.ManagedRuleSet, error) {
-	results := make([]network.ManagedRuleSet, 0)
+func expandWebApplicationFirewallPolicyManagedRuleSet(input []interface{}, d *pluginsdk.ResourceData) ([]webapplicationfirewallpolicies.ManagedRuleSet, error) {
```

Comment:
we always return a pointer and an error when an error is returned too:

```suggestion
func expandWebApplicationFirewallPolicyManagedRuleSet(input []interface{}, d *pluginsdk.ResourceData) (*[]webapplicationfirewallpolicies.ManagedRuleSet, error) {
```

Reply:
slice can function as a pointer in many instances, though it is not a pointer. I'll update it to a pointer to slice to make it the same as others.